### ISSUE TITLE
Error with num GPUs when using 0.8.0rc1

### ISSUE THREAD

**Issue by pamparana34** [PyTorchLightning/pytorch-lightning#2155]
I am trying to run my trainer while utilising 4 GPUs on my system.

I launch my trainer as follows:

```
trainer = Trainer(gpus=-1, distributed_backend='ddp')
```

This somehow goes into a tizzy and compares about some hyper parameter that was not even added by me. So the error is:

```
train_siamnet.py: error: unrecognized arguments: --gpus 4
```

The thing is that I did not pass this argument, so this is something probably added by PL somewhere down the line.

The model that I use and the launch script is as here:

https://github.com/PyTorchLightning/pytorch-lightning/issues/2069

**Comment by pamparana34** [comment]
So it seems one solution is to have the --gpus parameter exposed in the training file and basically ignore it from your end and let PL fill it while calling the script. Is that documented somewhere or is this not something I should be doing?

**Comment by zackcarson** [comment]
> So it seems one solution is to have the --gpus parameter exposed in the training file and basically ignore it from your end and let PL fill it while calling the script. Is that documented somewhere or is this not something I should be doing?

I'm also having this issue in 0.8.0rc1! Can you describe more how you fixed this?

**Comment by awaelchli** [comment]
The ddp mode now (since 0.8.0) launches your script several times. It expects you to have an argument parser with that gpus argument. It will call your script internally with 
python train_siamnet.py --gpus 4, right @williamFalcon ?

so the fix is to add this argument to your script and pass it to Trainer.

**Comment by awaelchli** [comment]
or use `backend="ddp_spawn"` which can be slower.

**Comment by Borda** [comment]
@williamFalcon is this solved in #2212?

**Comment by williamFalcon** [comment]
yes
